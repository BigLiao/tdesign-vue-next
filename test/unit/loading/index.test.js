import { mount } from '@vue/test-utils';
import Loading from '@/src/loading/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Loading', () => {
  // test props api
  describe(':props', () => {
    it(':default', () => {
      const wrapper = mount({
        render() {
          return <Loading loading={true}></Loading>;
        },
      });
      expect(wrapper.isEmpty()).toBe(false);
    });
    it(':delay', () => {
      const delay = 1000;
      const wrapper = mount({
        render() {
          return <Loading loading={true} delay={delay}></Loading>;
        },
      });
      setTimeout(() => {
        expect(wrapper.isEmpty()).toBe(false);
      }, delay);
    });
    it(':fullscreen', () => {
      const wrapper = mount({
        render() {
          return <Loading loading={true} fullscreen={true}></Loading>;
        },
      });
      expect(wrapper.isEmpty()).toBe(false);
    });
    it(':indicatorFunc', () => {
      const wrapper = mount({
        render() {
          return <Loading loading={true} indicator={() => '加载中...'}></Loading>;
        },
      });
      expect(wrapper.isEmpty()).toBe(false);
    });
    it(':preventScrollThrough', () => {
      const wrapper = mount({
        render() {
          return <Loading loading={true} preventScrollThrough={true} fullscreen={true}></Loading>;
        },
      });
      expect(wrapper.isEmpty()).toBe(false);
    });
    it(':size', () => {
      const wrapper = mount({
        render() {
          return <Loading loading={true} size={'large'}></Loading>;
        },
      });
      expect(wrapper.isEmpty()).toBe(false);
    });
    it(':text', () => {
      const wrapper = mount({
        render() {
          return <Loading loading={true} text={'正在加载...'}></Loading>;
        },
      });
      expect(wrapper.isEmpty()).toBe(false);
    });
    it(':wrap', () => {
      const wrapper = mount({
        render() {
          return <Loading loading={true}><div>this is loading component</div></Loading>;
        },
      });
      expect(wrapper.isEmpty()).toBe(false);
    });
  });
  // test slots
  describe('<slot>', () => {
    it('<indicator>', () => {
      const wrapper = mount(Loading, {
        render() {
          return <Loading loading={true}></Loading>;
        },
        scopedSlots: {
          indicator: '<i class="t-icon t-icon-loading"></i>',
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
