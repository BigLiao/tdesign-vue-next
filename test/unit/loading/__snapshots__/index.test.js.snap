// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading <slot> <indicator> 1`] = `
<div
  class="t-loading t-size-m t-loading-mask t-loading-lock v-transfer-dom"
>
  <svg
    class="t-loading__gradient t-icon-loading"
    height="1em"
    size="medium"
    version="1.1"
    viewBox="0 0 14 14"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <foreignobject
      height="12"
      width="12"
      x="1"
      y="1"
    >
      <div
        class="t-loading__gradient-conic"
      />
    </foreignobject>
  </svg>
  <!---->
</div>
`;
